torchmetrics/metric.py

```diff
@@ -247,6 +248,7 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
         cache = {attr: getattr(self, attr) for attr in self._defaults}
 
         # call reset, update, compute, on single batch
+        self._enable_grad = True  # allow grads for batch computation
```

Comment:
shouldn't we check whether this is enabled by `torch.is_grad_enabled`?

Reply:
This flag controls if we allow gradient to be passed through not torch or the user.
The only time we allow this is when the user computes the metric value on the current batch.

Every time we accumulate we want to disable this such that memory does not increase over time. 
That is actual problem this PR is solving, as we currently was not disabling it if user called `metric.update`.